### ISSUE TITLE
Unknown directives with typescript-mongodb and typescript-react-apollo

### ISSUE THREAD

**Issue by damianfrizzi** [dotansimha/graphql-code-generator#3117]
**Describe the bug**
When generating multiple files I can't use the `typescript-mongodb` together with `typescript-operations` / `typescript-react-apollo` plugin. Either you'll have missing directives or run into the error `typescript-mongodb` requires extension to be `.ts`!

**To Reproduce**
- Run the config file below with any documents and schema (containing at least one directive from typescript-mongodb)

1. My GraphQL schema:

```graphql
* Irelevant
```

2. My GraphQL operations:

```graphql
* Irelevant
```

3. My `codegen.yml` config file:

```yml
overwrite: true
schema: 'graphql/**/*.graphql'
documents: 'components/**/*.graphql'
generates:
  generated/graphql.db.ts:
    plugins:
      - add: // Generated File. Do not change it manually!
      - 'typescript'
      - 'typescript-mongodb'
  generated/graphql.server.ts:
    plugins:
      - add: // Generated File. Do not change it manually!
      - 'typescript'
      - 'typescript-resolvers'
    config:
      namingConvention:
        enumValues: keep
  generated/graphql.client.tsx:
    plugins:
      - add: // Generated File. Do not change it manually!
      - 'typescript'
      - 'typescript-operations'
      - 'typescript-react-apollo'
    config:
      withHooks: true
      withComponent: false
      withHOC: false
      skipTypename: true
      namingConvention:
        enumValues: keep
```

**Expected behavior**
It should be possible to use `typescript-mongodb` together with `typescript-operations` / `typescript-react-apollo`.

**Environment:**

- OS: 
- `@graphql-codegen/typescript-mongodb": "1.9.1"`
- `"@graphql-codegen/add": "1.9.1"`
- `"@graphql-codegen/cli": "1.9.1"`
- `"@graphql-codegen/typescript": "1.9.1"`
- `"@graphql-codegen/typescript-graphql-files-modules": "1.9.1"`
- `"@graphql-codegen/typescript-operations": "1.9.1"`
- `"@graphql-codegen/typescript-react-apollo": "1.9.1"`
- `"@graphql-codegen/typescript-resolvers": "1.9.1"`
- NodeJS: v10.16.0

**Additional context**
Related to https://github.com/dotansimha/graphql-code-generator/issues/985#issuecomment-444316509

**Comment by damianfrizzi** [comment]
Current setup to workaround the issue is to create a separate files for the mongo model definitions. Once the issue is resolved the files can be merged.

```yml
overwrite: true
generates:
  # Workaround https://github.com/dotansimha/graphql-code-generator/issues/3117
  generated/db.ts:
    schema: 'graphql/**/*.db.graphql'
    plugins:
      - add: import { Maybe } from './graphql.server';
      - 'typescript-mongodb'
    config:
      namingConvention:
        enumValues: keep
  generated/graphql.server.ts:
    schema: 'graphql/**/*.schema.graphql'
    plugins:
      - 'typescript'
      - 'typescript-resolvers'
    config:
      namingConvention:
        enumValues: keep
  generated/graphql.client.tsx:
    schema: 'graphql/**/*.schema.graphql'
    documents: 'components/**/*.graphql'
    plugins:
      - 'typescript'
      - 'typescript-operations'
      - 'typescript-react-apollo'
    config:
      withHooks: true
      withComponent: false
      withHOC: false
      skipTypename: true
      namingConvention:
        enumValues: keep
```

**Comment by dotansimha** [comment]
The codegen loads the schema globally when it's defined on the root of the documents, and loaded plugins are allowed to modify this schema instance (using `addToSchema` api). 

@damianfrizzi I tried to allow `.tsx` extension in ts-mongo, I think it should solve the issue (https://github.com/dotansimha/graphql-code-generator/pull/3190). Can you try the alpha version `1.9.2-alpha-bdfbf3eb.85`?

If it's still not working, can you please try to create a reproduction in a sandbox/repo?

**Comment by dotansimha** [comment]
Fixed in `v1.10.0`.
